azext_iot/tests/iothub/state/conftest.py

```diff
@@ -0,0 +1,523 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+from time import sleep
+import pytest
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.iothub.common import EndpointType
+from azext_iot.tests.settings import DynamoSettings
+from azext_iot.tests.generators import generate_generic_id
+from azext_iot.common.certops import create_self_signed_certificate
+from typing import Optional, TypeVar
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+SubRequest = TypeVar('SubRequest')
+Mark = TypeVar('Mark')
+
+
+MAX_RBAC_ASSIGNMENT_TRIES = 10
+USER_ROLE = "IoT Hub Data Contributor"
+
+
+cli = EmbeddedCLI()
+REQUIRED_TEST_ENV_VARS = ["azext_iot_testrg"]
+settings = DynamoSettings(req_env_set=REQUIRED_TEST_ENV_VARS)
+RG = settings.env.azext_iot_testrg
+
+
+def generate_hub_id() -> str:
+    return f"test-hub-{generate_generic_id()}"[:23]
+
+
+def generate_hub_depenency_id() -> str:
+    return f"testhubdep{generate_generic_id()}"[:24]
+
+
+def tags_to_dict(tags: str) -> dict:
+    result = {}
+    split_tags = tags.split()
+    for tag in split_tags:
+        kvp = tag.split("=")
+        result[kvp[0]] = kvp[1]
+    return result
```

Comment:
This would be a great utility function to have somewhere more central if we don't already. Just a thought.

Reply:
Looks like a copy from .../tests/deviceupdate/conftest.py....for re-usable test utilities like this, we can place them in a parent conftest (where conftest _should_ be mainly for test fixture definitions) or shared test utilities module.

Reply:
even more - I do not use this...